src/py/flwr/server/utils/validator.py

```diff
@@ -63,12 +64,21 @@ def validate_task_ins_or_res(tasks_ins_res: Union[TaskIns, TaskRes]) -> List[str
         ):
             validation_errors.append("non-anonymous consumer MUST provide a `node_id`")
 
-        # Legacy ServerMessage
-        if not tasks_ins_res.task.HasField("legacy_server_message"):
+        # Content check
+        has_fields = {
+            "sa": tasks_ins_res.task.HasField("sa"),
+            "legacy_server_message": tasks_ins_res.task.HasField(
+                "legacy_server_message"
+            ),
+        }
+        if sum(has_fields.values()) == 0:
```

Comment:
Let's make this more boring:
```suggestion
        if not (has_fields["sa"] or has_fields["legacy_server_message"]):
```